### PR TITLE
[PM-1935] Migrate vault lock screen to component library

### PR DESCRIPTION
**Pull request by Hinton**
## Ticket

https://bitwarden.atlassian.net/browse/PM-1935

## Objective

Migrate the vault lock screen to the component library with AnonLayout.

- **lock.component.html:** Added `[bitSubmit]` and `[formGroup]` binding, and `<bit-form-field>`
- **lock.component.ts:** Added a formGroup and converted submit to an arrow function
- **oss-routing.module.ts:** Added routing under AnonLayoutWrapper component

## Screencast

https://github.com/bitwarden/clients/assets/102181210/471d6e32-3f43-408f-80f2-aa11f16d80f3

### CODE REVIEW

**Review by djsmith85 — COMMENTED** [review]
Should the card-component be moved into libs/components? As it doesn't depend on anything Bitwarden/PasswordManager-specific it could easily be moved elsewhere. 

This would also make it easier to be re-used on web and desktop

**Comment by Hinton** [comment]
@djsmith85 it's not a component the CL team will support. It's strictly for sharing the container behavior for the auth pages.

**Comment by djsmith85** [comment]
> @djsmith85 it's not a component the CL team will support. It's strictly for sharing the container behavior for the auth pages and avoid inconsistencies which has already started to appear.

Then into libs/auth? Tools will need the `card.component` for the `send/access.component. Or the `card.component`will be part of the [AnonLayout](https://github.com/bitwarden/clients/pull/6424) which the SendAccess page will use?

**Comment by rr-bw** [comment]
@Hinton Is this still in being worked on? I'm happy to review once merge conflicts are resolved. No rush from me though.

**Review by rr-bw — CHANGES_REQUESTED** [review]
I believe we should use AnonLayout here, based on the [Figma design for the lock component](https://www.figma.com/design/81uGWWdQVAQfKDiR4qltoq/Email-Verification-[Create-account]?node-id=536-49011&t=FK0lR4AWGWiTONe6-0). 

Something like:
```typescript
import {
  AnonLayoutWrapperComponent,
  AnonLayoutWrapperData,
  LockIcon,
} from "@bitwarden/auth/angular";

...

{
  path: "",
  component: AnonLayoutWrapperComponent,
  children: [
    {
      path: "lock",
      children: [
        {
          path: "",
          component: LockComponent,
        },
      ],
      canActivate: [deepLinkGuard(), lockGuard()],
      data: {
        pageTitle: "yourVaultIsLocked",
        pageIcon: LockIcon,
      } satisfies AnonLayoutWrapperData,
    },
  ],
},
```

**Comment by Hinton** [comment]
@rr-bw this PR has been pending review for 8 months. I rebased it on request but I don't have the capacity in the forseeable future to allocate any resources to expanding the scope of the PR. The Auth team is welcome to take ownership of the PR.

**Comment by rr-bw** [comment]
@Hinton Yes I think that's best. I talked with Todd and Auth will take it from here.

**Review by rr-bw — DISMISSED** [review]

File: apps/web/src/app/auth/lock.component.ts [hunk 1]
```diff
@@ -1,18 +1,122 @@
...
 @Component({
   selector: "app-lock",
   templateUrl: "lock.component.html",
+  standalone: true,
+  imports: [SharedModule],
 })
 export class LockComponent extends BaseLockComponent {
+  formGroup = this.formBuilder.group({
+    masterPassword: ["", Validators.required],
+  });
```
Comment (jlf0dev) [inline apps/web/src/app/auth/lock.component.ts:41]: Can we change this to trigger on submit instead of the default on blur?

**Review by jlf0dev — CHANGES_REQUESTED** [review]

File: apps/web/src/app/auth/lock.component.ts (hunk 1 above)
Comment (danielleflinn) [reply apps/web/src/app/auth/lock.component.ts:41]: I'm good with this change!

**Comment by danielleflinn** [comment]
Design nit: Could we change the title to just "Your vault is locked"? 
It feels really long right now and I think entering the MP to unlock is implied.

**Review by jlf0dev — APPROVED** [review]
Thank you so much for shepherding this through and taking the time to verify the changes that were requested! You rock @rr-bw 👏

**Review by willmartian — APPROVED** [review]
LGTM!

**Pull request merged**